lib/CoreBot.js

```diff
@@ -568,10 +576,13 @@ function Botkit(configuration) {
                         (duration > this.task.timeLimit) && // timelimit is up
                         (lastActive > this.task.timeLimit) // nobody has typed for 60 seconds at least
                     ) {
-
-                        if (this.threads.timeout) {
+                        // if timeoutHandler is set then call it, otherwise follow the normal flow
+                        // this will not break others code, after the update
+                        if (this.timeOutHandler != null || this.timeOutHandler != undefined) {
```

Comment:
Also shouldn't this line be: `this.timeOutHandler != null && this.timeoutHandler != undefined` if it is null OR undefined, next line will fail. @benbrown